quartz-core/src/main/java/org/quartz/core/QuartzSchedulerThread.java

```diff
@@ -27,6 +27,8 @@
 import org.quartz.SchedulerException;
 import org.quartz.Trigger;
 import org.quartz.Trigger.CompletedExecutionInstruction;
+import org.quartz.impl.jdbcjobstore.JobStoreSupport;
```

Comment:
The direct dependencies on a given JobStore implementation is running afoul of intended abstraction and isolation of the code organization / design.

I think we need to consider a new abstraction for this fix that doesn't create a direct dependency on JobStoreSupport.